### PR TITLE
Shared action button component

### LINKED ISSUE #542: [UI Components] Action Button
Create a reusable action button component that allows for customization in the scope of how it is currently being implemented throughout the application (e.g. color, height, border, etc.)

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Replace all buttons with new shared Button component

## Type of change
Type: Non-breaking refactor

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #542 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by mxosman — COMMENTED** [review]
This is amazing, Ivan! Thank you so much for knocking down the first one on the list! Love your refactor & decisions. One request -- can we move all of these to the `common` repo? Sorry I didn't specify this before - but ultimately, all of these UI components will live in the `common` repo. We can include them in `common/components` folder.

File: publisher/src/components/DataUpload/DataUpload.styles.tsx [hunk 1]
```diff
@@ -196,127 +196,127 @@ export const ErrorWarningButtonWrapper = styled(ButtonWrapper)`
...
+//
+//   a {
+//     ${typography.sizeCSS.small};
+//     width: fit-content;
+//     text-decoration: none;
+//     color: ${palette.solid.blue};
+//     display: flex;
+//     align-items: center;
+//   }
+// `;
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/DataUpload/DataUpload.styles.tsx]: [b51d2ec](https://github.com/Recidiviz/justice-counts/pull/557/commits/b51d2ecc8a3d6fb16edea109608476dc3d03a39a)

File: publisher/src/components/shared/Button/Button.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+} from "./types";
+
+export const Button = styled.div<{
+  buttonColor?: ButtonColor;
+  labelColor?: ButtonLabelColor;
+  borderColor?: ButtonBorderColor;
+  size?: ButtonSize;
+  enabledDuringOnboarding?: boolean;
+  disabled?: boolean;
+  hasNoSidePadding?: boolean;
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/shared/Button/Button.styled.tsx]: [b51d2ec](https://github.com/Recidiviz/justice-counts/pull/557/commits/b51d2ecc8a3d6fb16edea109608476dc3d03a39a)

File: publisher/src/components/shared/Button/Button.styled.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  ${({ labelColor, buttonColor, enabledDuringOnboarding }) => {
+    if (labelColor === "blue")
+      return `
+        color: ${palette.solid.blue};
+      `;
+    if (labelColor === "red")
+      return `
+        color: ${palette.solid.red};
+      `;
+    // maybe remove button color later if there would be colored buttons without white label
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/shared/Button/Button.styled.tsx:81]: [b51d2ec](https://github.com/Recidiviz/justice-counts/pull/557/commits/b51d2ecc8a3d6fb16edea109608476dc3d03a39a)

File: publisher/src/components/shared/Button/types.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+export type ButtonSize = "medium";
+
+// export type ButtonType =
+//   | "light-border"
+//   | "border"
+//   | "borderless"
+//   | "blue"
+//   | "green"
+//   | "red"
+//   | "orange";
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply publisher/src/components/shared/Button/types.ts]: [b51d2ec](https://github.com/Recidiviz/justice-counts/pull/557/commits/b51d2ecc8a3d6fb16edea109608476dc3d03a39a)

**Comment by ivan-kishko** [comment]
> This is amazing, Ivan! Thank you so much for knocking down the first one on the list! Love your refactor & decisions. One request -- can we move all of these to the `common` repo? Sorry I didn't specify this before - but ultimately, all of these UI components will live in the `common` repo. We can include them in `common/components` folder.

[87b8c5c](https://github.com/Recidiviz/justice-counts/pull/557/commits/87b8c5c2c9c51e3e513322dcd7e27589fa801090)

**Comment by lilidworkin** [comment]
This looks awesome @ivan-kishko ! For this PR and the other component PRs, I'm going to let @mxosman take the lead on reviewing / accepting. When he accepts feel free to merge!

**Review by mxosman — APPROVED** [review]
Thank you so much, Ivan! Looks great to me!
🙏 🙌

**Comment by mxosman** [comment]
Ah - shoot. I caused some conflicts here - let me resolve them and push up a commit since there were some changes to some of the review pages.

**Pull request merged**